### PR TITLE
Reconnect Fix

### PR DESCRIPTION
**Pull request by amngupta**
- Call the disconnect function
- Call the connect function

### CODE REVIEW

File: sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java [hunk 1]
```diff
@@ -400,51 +400,47 @@ private void establishWebsocketConnection() throws IOException
         }
         establishConnectionThread = null;
 
-        establishConnectionThread = new Thread()
+        establishConnectionThread = new Thread(() ->
```
Comment (bcorne) [inline sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java]: You shouldn't use lambdas or any other java 8 feature, the source and bytecode should remain compatible with java 7 for now.

File: sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java [hunk 2]
```diff
@@ -915,43 +1033,51 @@ private HttpClient getHttpClient() {
...
             LOGGER.debug("pong received " + lastPingAck);
         }
-        else if ("reconnect_url".equals(object.get("type").getAsString())) {
-            webSocketConnectionURL = object.get("url").getAsString();
-            LOGGER.debug("new websocket connection received " + webSocketConnectionURL);
-        }
+        //        else if ("reconnect_url".equals(object.get("type").getAsString())) {
+        //            webSocketConnectionURL = object.get("url").getAsString();
+        //            LOGGER.debug("new websocket connection received " + webSocketConnectionURL);
+        //        }
```
Comment (bcorne) [inline sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java]: You can safely remove the commented block, git will allow to retrieve it through the history if we ever need to bring it back

File: sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java (hunk 1 above)
Comment (amngupta) [inline sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java]: Fixed

File: sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java [hunk 3]
```diff
@@ -334,43 +348,50 @@ private void connectImpl() throws IOException
...
 
-        try {
-            websocketSession = client.connectToServer(new Endpoint()
-            {
-                @Override
-                public void onOpen(Session session, EndpointConfig config)
-                {
-                    session.addMessageHandler(handler);
-                }
+        establishConnectionThread = new Thread() {
```
Comment (bcorne) [inline sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java]: looks a bit complex here: I don't see the point to delegate the websocket connection to a thread. I would keep the old way to start the connection to the server and maybe use a locking mechanism to ensure only one connection attempt is made at the same time. I missed this yesterday with the code reformatting.

Comment (amngupta) [inline sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java]: Without threading it I am getting a 
`
Caused by: javax.websocket.DeploymentException
	at org.glassfish.tyrus.client.ClientManager.connectToServer(ClientManager.java:380)
	at com.ullink.slack.simpleslackapi.impl.SlackWebSocketSessionImpl.establishWebsocketConnection(SlackWebSocketSessionImpl.java:402)
	... 4 more
Caused by: java.lang.InterruptedException
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly`

That is the reason I introduced a new thread to handle the establishWebsocketConnection behaviour

Comment (amngupta) [inline sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java]: Fixed

**Pull request closed without merging**